wapitiCore/net/web.py

```diff
@@ -348,82 +348,67 @@ def _encoded_keys(params):
 
     def __repr__(self):
         if self._get_params:
-            buff = "{0} {1} ({2})".format(self._method, self.url, self._link_depth)
+            buff = f"{self._method} {self.url} ({self._link_depth})"
         else:
-            buff = "{0} {1} ({2})".format(self._method, self._resource_path, self._link_depth)
+            buff = f"{self._method} {self._resource_path} ({self._link_depth})"
 
         if self._post_params:
-            buff += "\n\tdata: {}".format(self.encoded_data.replace("\n", "\n\t"))
+            buff += "\n\tdata: " + self.encoded_data.replace("\n", "\n\t")
         if self._file_params:
-            buff += "\n\tfiles: {}".format(self.encoded_files)
+            buff += "\n\tfiles: " + self.encoded_files
 
         return buff
 
     def http_repr(self, left_margin="    "):
         rel_url = self.url.split('/', 3)[3]
-        http_string = "{3}{0} /{1} HTTP/1.1\n{3}Host: {2}\n".format(
-            self._method,
-            rel_url,
-            self._hostname,
-            left_margin
-        )
+        http_string = f"{left_margin}{self._method} /{rel_url} HTTP/1.1\n{left_margin}Host: {self._hostname}\n"
 
         if self._referer:
-            http_string += "{}Referer: {}\n".format(left_margin, self._referer)
+            http_string += f"{left_margin}Referer: {self._referer}\n"
 
         if self._file_params:
             boundary = "------------------------boundarystring"
-            http_string += "{}Content-Type: multipart/form-data; boundary={}\n\n".format(left_margin, boundary)
+            http_string += f"{left_margin}Content-Type: multipart/form-data; boundary={boundary}\n\n"
             for field_name, field_value in self._post_params:
                 http_string += (
-                    "{3}{0}\n{3}Content-Disposition: form-data; "
-                    "name=\"{1}\"\n\n{3}{2}\n"
-                ).format(boundary, field_name, field_value, left_margin)
+                    f"{left_margin}{boundary}\n{left_margin}Content-Disposition: form-data; "
+                    "name=\"{field_name}\"\n\n{left_margin}{field_value}\n")
             for field_name, field_value in self._file_params:
                 http_string += (
-                    "{3}{0}\n{3}Content-Disposition: form-data; name=\"{1}\"; filename=\"{2}\"\n\n"
-                    "{3}{4}\n"
-                ).format(
-                    boundary,
-                    field_name,
-                    field_value[0],
-                    left_margin,
-                    field_value[1].decode(errors="replace").replace("\n", "\n" + left_margin).strip()
+                    f"{left_margin}{boundary}\n{left_margin}Content-Disposition: form-data; "
+                    f"name=\"{field_name}\"\n\n{left_margin}{field_value}\n"
```

Comment:
I did not check in detail, but you may have forgotten the `filename=\"{2}\"` section.
And `field_value[1].decode(errors="replace").replace("\n", "\n" + left_margin).strip()` has been replaced with `field_value`